OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -355,14 +356,8 @@ void PopulateAssetList()
 			assetList.RemoveChildren();
 			availableShps.Clear();
 
-			// TODO: This is too slow to run in the main thread
-			// var files = AssetSource != null ? AssetSource.AllFileNames() :
-			// GlobalFileSystem.MountedFolders.SelectMany(f => f.AllFileNames());
-			if (assetSource == null)
-				return;
-
-			var files = assetSource.Contents.OrderBy(s => s);
-			foreach (var file in files)
+			var files = assetSource != null ? assetSource.Contents : Game.ModData.ModFiles.MountedPackages.SelectMany(f => f.Contents);
```

Comment:
Can you tack a `.Distinct()` onto the end here, please? Otherwise you get duplicate entries in the `all packages` view for stuff where we have replacements in the bits directory.